api/envoy/extensions/quic/server_preferred_address/v3/fixed_server_preferred_address_config.proto

```diff
@@ -16,20 +18,51 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#extension: envoy.quic.server_preferred_address.fixed]
 
 // Configuration for FixedServerPreferredAddressConfig.
+// [#next-free-field: 7]
 message FixedServerPreferredAddressConfig {
   // [#comment:TODO(danzh2010): discuss with API shepherds before removing WiP status.]
 
   option (xds.annotations.v3.message_status).work_in_progress = true;
 
   oneof ipv4_type {
     // String representation of IPv4 address, i.e. "127.0.0.2".
+    // The listener's port will be used.
     // If not specified, none will be configured.
     string ipv4_address = 1;
+
+    // The IPv4 address to advertise to clients for Server Preferred Address.
+    config.core.v3.SocketAddress ipv4_address_and_port = 3;
```

Comment:
The string version does not allow setting the port. I think it was done this way because we don't yet support multiple sockets on a quic listener; it has to get the original and spa traffic on the same socket, bound to `0.0.0.0:443` or similar, so the SPA address must use the same port, and so it isn't allowed to be configured here.

I am hoping to allow multiple sockets in the future, but this is what we have for now.

But with DNAT, the client's view of the destination may not be what the server receives, so if the DNAT also does port translation, we can now support having a different port in the address advertised to clients.

Reply:
We had a string version because we don't want to make port part of the config as the port is supposed to be the same as the listening port.

With DNAT do we need to advertise a different port in transport parameter?

Reply:
Since you added a new knob to allow a different port. I think we can keep the old behavior of the string form with a comment "if it is desired to advertise a different port, use the struct form".

As to the new knob, how about defining a struct 
```
DNatAddressPair {
// Port will be used.
config.core.v3.SocketAddress address_to_advertise;
// Only the IP part. Different port is not supported yet. 
string dnat_internal_address;
};
```

And then
```
 oneof ipv4_type {
    // String representation of IPv4 address, i.e. "127.0.0.2".
    // The listener's port will be used.
    // If not specified, none will be configured.
    string ipv4_address = 1;

    DNatAddressPair ipv4_addresses_with_dnat = 2;
}
```

Reply:
Given that the current restriction on the port being equal is an implementation restriction, that may be alleviated in the future, would it be better to use `SocketAddress` for both addresses, and document that the port is currently ignored, so that we don't have to change proto in the future to allow setting the port if/when that happens?

Reply:
SG. And please also emit an error during config validation.

Reply:
I pushed a commit with what I think we've agreed on for the config. @danzh2010 @abeyad can you please take a look and confirm this looks roughly correct; then I'll make the related code changes (and add the validation that the port == 0 in the cases where it isn't/can't be used).